pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -772,6 +772,8 @@ public synchronized void close()
       }
     }
     doClose();
+    _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId, ServerGauge.UPSERT_PRIMARY_KEYS_COUNT,
+        0L);
```

Comment:
should we do this during stop() instead?

Reply:
I guess we want to finish all operations then reset the gauge, so here should be the right place

Reply:
Suggest adding some comments explaining why we reset the gauge here